src/commands/flow/internal/spec/allFlowsCompleted.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { allFlowsCompleted } from '..'
+
+describe('allFlowsCompleted', () => {
+  it('should return true, if all flows are completed', () => {
+    const flows = {
+      flow1: {
+        jobs: [{ location: 'job', status: 'success' as const }],
+        predecessors: ['flow4']
+      },
+      flow2: {
+        jobs: [{ location: 'job', status: 'failure' as const }],
+        predecessors: ['flow5', 'flow4']
+      },
+      flow3: {
+        jobs: [{ location: 'job', status: 'success' as const }],
+        predecessors: []
+      }
+    }
+
+    const { completed } = allFlowsCompleted(flows)
+
+    expect(completed).toEqual(true)
+  })
+
+  it('should return false, if all flows are not completed', () => {
+    const flows = {
+      flow1: {
+        jobs: [{ location: 'job', status: 'success' as const }],
+        predecessors: ['flow4']
+      },
+      flow2: {
+        jobs: [{ location: 'job' }],
+        predecessors: ['flow5', 'flow4']
+      },
+      flow3: {
+        jobs: [{ location: 'job', status: 'success' as const }],
+        predecessors: []
+      }
+    }
+
+    const { completed } = allFlowsCompleted(flows)
+
+    expect(completed).toEqual(false)
+  })
+
+  it('should return true, if all flows are completed successfully', () => {
+    const flows = {
+      flow1: {
+        jobs: [{ location: 'job', status: 'success' as const }],
+        predecessors: ['flow4']
+      },
+      flow2: {
+        jobs: [{ location: 'job', status: 'success' as const }],
+        predecessors: ['flow5', 'flow4']
+      },
+      flow3: {
+        jobs: [{ location: 'job', status: 'success' as const }],
+        predecessors: []
+      }
+    }
+
```

Comment:
we also need a test case when `should return true if all flows failed`

Reply:
added